scripts/simplify_network.py

```diff
@@ -354,37 +375,80 @@ def split_links(nodes):
                 n, busmap, costs, config, connection_costs_per_link, buses
             )
 
-            all_links = [i for _, i in sum(links, [])]
+            # TODO revise a variable name for `links` variable
+            # `links` is a list which contains dc-relevant graph elements
+            all_dc = [i for _, i in sum(links, [])]
+            all_links = list(set(n.links.index).intersection(all_dc))
+            all_dc_lines = list(set(n.lines.index).intersection(all_dc))
+
+            # p_max_pu = config["links"].get("p_max_pu", 1.0)
+            all_dc_lengths = pd.concat(
+                [n.links.loc[all_links, "length"], n.lines.loc[all_dc_lines, "length"]]
+            )
+            name = all_dc_lengths.idxmax() + "+{}".format(len(all_dc) - 1)
+
+            # HVDC part is represented as "Link" component
+            if dc_as_links:
+                p_max_pu = config["links"].get("p_max_pu", 1.0)
+                lengths = n.links.loc[all_links, "length"]
+                length = sum(
+                    n.links.loc[[i for _, i in l if _ == "Link"], "length"].mean()
+                    for l in links
+                )
+                p_nom = min(
+                    n.links.loc[[i for _, i in l if _ == "Link"], "p_nom"].sum()
+                    for l in links
+                )
+                underwater_fraction = sum(
+                    lengths
+                    / lengths.sum()
+                    * n.links.loc[all_links, "underwater_fraction"]
+                )
+            # HVDC part is represented as "Line" component
+            elif dc_as_lines:
+                p_max_pu = config["lines"].get("p_max_pu", 1.0)
+                lengths = n.lines.loc[all_dc_lines, "length"]
+                length = lengths.sum() / len(lengths) if len(lengths) > 0 else 0
+                p_nom = (
+                    n.lines.loc[all_dc_lines]
+                    .apply(
+                        lambda x: x["v_nom"] * n.line_types.i_nom[x["type"]],
+                        axis=1,
+                        result_type="reduce",
+                    )
+                    .min()
+                )
```

Comment:
Theoretically, there should be the column s_nom that contains the nominal capacity of the line: is it filled properly?
If so, you may use that column here.

However, you raised an interesting point: in `simplify_network_to_380` we still assume that all lines are AC, it may be interesting to debug what happens there with DC lines.
In particular, the num_parallel is most likely calculated in a wrong way

Reply:
Mmm, you are absolutely right: there is `s_nom` in `n.lines`, and it's value is exactly 3 times lower as compared with U*I used above. It feels like there is in fact some discrapancy with `num_parallel`

Reply:
Fixed